include/battle.h

```diff
@@ -675,6 +675,24 @@ struct BattleStruct
         typeArg = gBattleMoves[move].type;                            \
 }
 
+#define GET_MOVE_CALLED(move, targetArg)                              \
+{                                                                     \
+    switch (gBattleMoves[move].effect)                                \
+    {                                                                 \
+    case EFFECT_COPYCAT:                                              \
```

Comment:
Copycat needs its own case here; it repeats the last move used in the battle, by any battler, including the user - it doesn't have a target like Mirror Move does, which gets messy in a double battle. To figure what the copied move might be you have to account for turn order.